### DISCUSSION TITLE
Platform specific directories (instead of ~/.datasette-app)

### DISCUSSION THREAD

**Discussion by kfdm** [simonw/datasette-app#124] [Ideas]
A lot of applications have a habit of creating an `~/.<appname>` directory, but most platforms typically have a preferred location for files. For example, on Linux distros usually that should be `~/.config/<appname>` or `~/.cache/<appname>` 
https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html

On MacOS, that would often be `~/Library/Application Support/<appname>` and `~/Library/Cache/<appname>` 

Windows has an `Application Data` folder.

For Python, there are libraries like https://pypi.org/project/appdirs/ that handle some of this, though I would imagine for javascript there are similar libraries

**Comment by simonw** [comment]
Electron has mechanisms for finding these which I thought about using - but as a macOS user I can never remember where to look for this stuff!

I'm hoping to help people who want to learn how to use CLI tools by showing them how to add specific commands - like `datasette` and `sqlite-utils` - to their path, see #92. That feels even more weird to me if the binaries they are executing live somewhere in Application Support.

So not using Application Support is currently a deliberate design decision. I'm ready to be convinced it's the wrong call though!

**Reply by simonw** [reply]
I've not yet thought about how this should best work on Windows or Linux - I've just considered macOS.

**Reply by kfdm** [reply]
Took a while for me to think of a response. If you're writing tools for yourself, then putting the files wherever is easiest is fine. If you're writing tools for others, it feels like you have a bit more responsibility to respect the conventions of the platforms you are targeting.
